Sources/AWSLambdaRuntimeCore/LambdaRunner.swift

```diff
@@ -34,19 +35,34 @@ extension Lambda {
         /// Run the user provided initializer. This *must* only be called once.
         ///
         /// - Returns: An `EventLoopFuture<LambdaHandler>` fulfilled with the outcome of the initialization.
-        func initialize(logger: Logger, factory: @escaping HandlerFactory) -> EventLoopFuture<Handler> {
+        func initialize(serviceLifecycle: ServiceLifecycle, logger: Logger, factory: @escaping HandlerFactory) -> EventLoopFuture<Handler> {
             logger.debug("initializing lambda")
             // 1. create the handler from the factory
             // 2. report initialization error if one occured
             let context = InitializationContext(logger: logger,
                                                 eventLoop: self.eventLoop,
+                                                serviceLifecycle: serviceLifecycle,
                                                 allocator: self.allocator)
             return factory(context)
-                // Hopping back to "our" EventLoop is importnant in case the factory returns a future
+                // Hopping back to "our" EventLoop is important in case the factory returns a future
                 // that originated from a foreign EventLoop/EventLoopGroup.
                 // This can happen if the factory uses a library (let's say a database client) that manages its own threads/loops
                 // for whatever reason and returns a future that originated from that foreign EventLoop.
                 .hop(to: self.eventLoop)
+                .flatMap { (handler) in
+                    let promise = self.eventLoop.makePromise(of: ByteBufferLambdaHandler.self)
+                    // after we have created the LambdaHandler we must now start the services.
+                    // in order to not have to map once our success case returns the handler.
+                    serviceLifecycle.start { (error) in
```

Comment:
I wonder if we should add sugar in the lifecycle NIO compact module to make this kind of integration easier. wdyt?

Reply:
https://github.com/swift-server/swift-service-lifecycle/pull/72